src/olgm/herald/feature.js

```diff
@@ -251,3 +281,31 @@ olgm.herald.Feature.prototype.handleGeometryReplace_ = function() {
   keys.push(this.geometryChangeKey_);
   this.handleGeometryChange_();
 };
+
+
+/**
+ * @private
+ */
+olgm.herald.Feature.prototype.handleChange_ = function() {
+  // detect if the styles have changed, if so, update the gmap feature
+  var style = this.cacheStyle_(olgm.getStyleOf(this.feature_));
+  if (goog.json.serialize(style) !== goog.json.serialize(this.cachedStyle_)) {
```

Comment:
>I had to add goog.json, because goog.object.equals does not deal with deep comparisons well and ol.style.Style does not offer its own equals comparator.

Let's wait to see if changing the `change` logic stuff could the trick, i.e. listening to `change` on the Style object itself.  We wouldn't need the above comparison.

If we end up requiring it still, then we would need to remove the use of `goog.json`.  We removed every dependencies on the Closure Library with the exception of `goog.require` and `goog.provide`, which should be replaced in the future once we change to `require` and `export` instead (and go to Webpack or something like that).  So, if that's the case, let's simply copy the source code and implement it as `olgm.json` instead.